### ISSUE TITLE
bug: Stencil 2.16.0 does not build loader folder 

### ISSUE THREAD

**Issue by Tallyb** [ionic-team/stencil#3403]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.16.0

### Current Behavior

In 2.15.0 a typical folder structure of stencil dist folder (with type 'dist') looks something like: 
- collection
- design-system
- types
- cjs
- esm
- loader
and in addition the top level folder included an index and an index.cjs files. 
In 2.16.0 only the design-system and types folders exist and no files at the loader. 

### Expected Behavior

Not sure if this is a regression bug or intentional. If the latter, I would expect to see a migration guide if loading via the loader. 

### Steps to Reproduce

build with 2.16.0
Edited: run `stencil build --dev`

### Code Reproduction URL

https://github.com/Tallyb/stencil-one

**Comment by rwaskiewicz** [comment]
Hey @Tallyb 👋 

I tried to reproduce this by spinning up two identical stencil projects, one with v2.15.2 and one with v2.16.0. I built each project, then I `diff`'ed the two directories (excluding `.git`, `node_modules` and `www`). I'm not seeing any different behavior in how the loader directory is generated for either of the following `dist` output target configurations:

```json
    {
      type: 'dist',
      esmLoaderPath: '../loader',
    },
```
```json
    {
      type: 'dist',
    },
```

Would it be possible for you to create a minimal reproduction case for the team to take a look at? Ideally, if we could have an example of how it behaves on v2.15.2 on one commit/branch, and another example of this reported behavior using v2.16.0 on another, that'd be much appreciated 🙏

**Comment by Tallyb** [comment]
@rwaskiewicz - can you try and build the repo mentioned above? When building with 2.16.0 the loader folder is missing.

**Comment by rwaskiewicz** [comment]
Hey @Tallyb, I've tried to do reproduce this myself myself with the default component starters, and was unable to. 

In building your project with v2.16.0 and v2.15.2, I can see a difference in the output of running `npm run build` and looking at `dist`. I did this at 5762b3827f3dbd0146dea35e8bef7f1cc81f5e2d (with v2.16.0) and 6e9c9ced82b2699860efaeaf907b03af2b5315fa (it's parent, with v2.15.2).

To move this further, a minimal reproduction case would be ideal/required.  As someone who knows your stencil-one repo the best, it'd be immensely helpful if you could create a new starter library with `npm init stencil@latest component my-library` and attempt to reproduce from there, pulling in bits and pieces from your stencil-one repository until we can reproduce the issue faithfully.

**Comment by Tallyb** [comment]
@rwaskiewicz  - Ok, this happens if you use the --dev in the stencil build command. Not sure why and what it means.

**Comment by alicewriteswrongs** [comment]
@Tallyb hello! I'm just jumping in because I believe I've narrowed down what the problem is.

Could you confirm that, on any version of stencil < 2.16.0, building without the `--esm` flag reproduces the issue?

This flag was in a sort of half-supported state, and that half-support was recently broken, which I believe is the root cause of your issue.

**Comment by rwaskiewicz** [comment]
@Tallyb Thanks to @alicewriteswrongs, we believe we have a fix for this issue. Can you please install a pre-release of Stencil that has just been cut to confirm? In running the following on our side, the loader directory appears to now be generated correctly in the pre-release v2.16.1-0 of Stencil
```
npm install @stencil/core@2.16.1-0
rm -rf dist
npm run build
```

**Comment by Tallyb** [comment]
@rwaskiewicz  - I can confirm it is producing similar output with 2.16.1-0

**Comment by rwaskiewicz** [comment]
Thanks @Tallyb, we've released an official version of v2.16.1 that includes the fix!
